python-client/giskard/scanner/robustness/base_detector.py

```diff
@@ -65,6 +65,7 @@ def _detect_issues(
     ) -> Sequence[Issue]:
         num_samples = self.num_samples or _get_default_num_samples(model)
         output_sensitivity = self.output_sensitivity or _get_default_output_sensitivity(model)
+        threshold = self.threshold or _get_default_threshold(model)
```

Comment:
You are right, threshold should be set to `None` by default

Reply:
I'd have a `self._threshold` property and a threshold getter that would take self.threshold if defined otherwise will apply the _get_default_threshold logic. Like this in the rest of the code we just refer to the `self.threshold` getter

Reply:
actually never mind, `model` isn't a class property, so getter won't work